### PR TITLE
Add submission_events() and the ability to fetch all person aliases (without specifying a person)

### PR DESCRIPTION
**Pull request by lumisota**

### CODE REVIEW

File: ietfdata/datatracker.py [hunk 1]
```diff
@@ -276,6 +276,16 @@ class Submission:
     words           : Optional[int]
 
 
+@dataclass(frozen=True)
+class SubmissionEvent:
+    by              : Optional[PersonURI]
+    desc            : str
+    id              : int
+    resource_uri    : str
```
Comment (csperkins) [inline ietfdata/datatracker.py]: Should this be a `SubmissionEventURI` type? The goal is to move away from URIs as strings.

File: ietfdata/datatracker.py [hunk 2]
```diff
@@ -673,8 +683,10 @@ def person_from_email(self, email_addr: str) -> Optional[Person]:
             return None
 
 
-    def person_aliases(self, person: Person) -> Iterator[PersonAlias]:
-        url = "/api/v1/person/alias/?person=" + str(person.id)
+    def person_aliases(self, person: Optional[Person] = None) -> Iterator[PersonAlias]:
+        url = "/api/v1/person/alias/"
+        if person is not None:
+            url += "?person=" + str(person.id)
```
Comment (csperkins) [inline ietfdata/datatracker.py]: Would it be better to leave `person_aliases()` unchanged, and add `people_aliases(self, since, until)` as well?

Comment (lumisota) [reply ietfdata/datatracker.py]: The /person/alias endpoint doesn't have timestamps (or allow filtering by timestamp), unfortunately. 

Originally, my thought was to fetch aliases independently of person objects, but without being able to filter by timestamp, this isn't practical (you'd have to fetch them all every time). So, I don't think I need this change anymore, but I'm unsure if it's worth including for other uses.

File: ietfdata/datatracker.py (hunk 1 above)
Comment (lumisota) [reply ietfdata/datatracker.py]: Yes; changed now in 7270acc

Comment (csperkins) [reply ietfdata/datatracker.py]: Thanks – also needs a `submission_event(self, submission_event_uri)` method.

File: ietfdata/datatracker.py (hunk 2 above)
Comment (csperkins) [reply ietfdata/datatracker.py]: I'd leave it out if we can't filter them. Any change to a person that creates a new alias will presumably show up as an update to the Person object anyway, so it's possible to tell when to re-fetch the aliases for a person?

File: ietfdata/datatracker.py (hunk 1 above)
Comment (lumisota) [reply ietfdata/datatracker.py]: Done -- added in the latest commits, along with a ```test_submission_event()``` method in ```test_datatracker.py```.

File: ietfdata/datatracker.py (hunk 2 above)
Comment (lumisota) [reply ietfdata/datatracker.py]: Agreed; rolled this change back.

I presume that a new alias would result in the person object's timestamp being updated, but I'd need to experiment to check that.

File: ietfdata/datatracker.py [hunk 3]
```diff
@@ -976,6 +993,39 @@ def submission(self, submission_uri: SubmissionURI) -> Optional[Submission]:
...
 
+    def submission_event(self, event_uri: SubmissionEventURI) -> Optional[SubmissionEvent]:
+        return self._retrieve(event_uri, SubmissionEvent)
+
+
+    def submission_events(self,
+                        since      : str = "1970-01-01T00:00:00",
+                        until      : str = "2038-01-19T03:14:07",
+                        by         : PersonURI            = None,
+                        submission : SubmissionURI        = None,
```
Comment (csperkins) [inline ietfdata/datatracker.py]: Should these be `Optional[]` - there are `is not None` tests below?

Comment (lumisota) [reply ietfdata/datatracker.py]: Yes; fixed in 6fa463a

Comment (csperkins) [reply ietfdata/datatracker.py]: Thanks!

**Pull request merged**